### DISCUSSION TITLE
How to run script on startup

### DISCUSSION THREAD

**Discussion by Zosu** [gtxaspec/wz_mini_hacks#628] [Q&A] [answered]
Hi,
I have a script that I want to run on the camera on reboot. It runs fine when I manually execute it, but I cannot seem to get it to run automatically after a reboot. I've tried the suggestions in #571 but I still can't seem to get it to work. What's the right way to do this?

**Comment by virmaior** [comment, marked as answer]
if you want it to run as part of the startup process place it `/opt/wz_mini/etc/network.d` and name it s99 or something like that so that it runs after everything else.

**Comment by Zosu** [comment]
Thanks, that seems to do the trick!

**Comment by archandanime** [comment]
@Zosu `/opt/wz_mini/etc/rc.local.d` works too and this directory is among "reserved" directories/files that survive after upgrade.
Source: [upgrade-run.sh](https://github.com/gtxaspec/wz_mini_hacks/blob/348ddac0caa3f5619add0c49649e053f5c6d39df/SD_ROOT/wz_mini/bin/upgrade-run.sh)
```
echo "Backup user config"
cp /opt/wz_mini/wz_mini.conf /opt/Upgrade/preserve/
cp -r /opt/wz_mini/etc/configs /opt/Upgrade/preserve/
cp -r /opt/wz_mini/etc/ssh /opt/Upgrade/preserve/
cp -r /opt/wz_mini/etc/wireguard /opt/Upgrade/preserve/
cp -r /opt/wz_mini/etc/rc.local.d /opt/Upgrade/preserve/
cp -r /opt/wz_mini/etc/cron /opt/Upgrade/preserve/
cp -r /opt/wz_mini/etc/resolv.dnsmasq /opt/Upgrade/preserve/
```
Scripts in this directory run after network is connected, so it really depends on your use case to choose the right directory.
Source: [wz-init.sh](https://github.com/gtxaspec/wz_mini_hacks/blob/348ddac0caa3f5619add0c49649e053f5c6d39df/SD_ROOT/wz_mini/etc/init.d/wz_init.sh)
```
#init.d/ = early boot, before inittab is run
#rc.d/ = runs after /linuxrc, but before app_init.sh
#network.d/ runs after app_init.sh, and after wlan hw is ready
#rc.local.d/ = runs after app_init.sh and network has acquired an address
```
